### DISCUSSION TITLE
Crazt bolt 1.1 compatibility with Readytosky 30A ESC

### DISCUSSION THREAD

**Discussion by Xinlyu123** [bitcraze/discussions#128] [Q&A] [answered]
Hi! I am a e beginner on drone and I am currently making drone with crazy bolt 1.1. However, the motors only spin for a while and stop, then the ESC start re-calibrating. Does anyone knows what reason causes this? 

Video is more than a thousand words: [https://youtube.com/shorts/nBswJ98YO9I?feature=share](url)

ESC: [https://www.amazon.com/Readytosky-Electronic-Controller-Helicopter-Quadcopter/dp/B09G5WFXSV/ref=sr_1_3?crid=1QCHZHHH2F8RF&keywords=ready+to+sky+esc&qid=1657032008&sprefix=%2Caps%2C129&sr=8-3#customerReviews](url)

My guesses:
1. ESC not calibrated?
2. Different protocol? Bolt use PWM/Oneshot125 protocol, and this ESC's protocol is not listed on any web page?

**Comment by tobbeanton** [comment, marked as answer]
I'm a bit unsure here, not much information to be found. Looking at the specs:
- 4, Supports up to 621Hz refresh rates up to the throttle signal, compatible with a variety of flight control (Note: More than 500Hz signal of the throttle throttle are all non-standard signal)

I'm interpreting this as it only supports PWM so make sure to use this. Else I think it might be your other question mark, that it needs some form of calibration.

**Reply by Xinlyu123** [reply]
Firstly, I want to thank you for the reply. 
Are you able to watch that video I posted? The problem is the motor only spin for 1 or 2 circles then it stops, and the throttle rate shows on the cfclient only go up to 79.3%. Also, the drone works fine with 2 motors, once I plug 4 motors the issue occurs
The crazybolt 1.1, it supports PWM and Oneshot125. I am wondering if I need to change the setting somewhere to make the protocol of bolt to PWM. Or it is already PWM protocol once I connect the bolt and ESCs.

**Reply by tobbeanton** [reply]
I have watched the video and I must say it is the first time I see that type of behavior. If you are running the latest release 2022.05 the default motor signal protocol is PWM @400Hz which should work with your ESC.

Maybe the motors and ESC generate some form of noise in the system that corrupts the motor signal. That would make sense as it works with 2 motors and not 4. However you don't have any props on so the loading should be small...

In the cfclient there is the Parameters tab. Connect to the running bolt and find param group **motorPowerSet**. The sub-parameter **enable** will override the motor output value if set to one and you can use **m1-m4** to output this value to respective motor. Try e.g. to set them to 10000 one by one and see how the motors behave.

**Reply by Xinlyu123** [reply]
I think I fixed the issue thanks for your help, I really appreciate it!!

**Reply by Xinlyu123** [reply]
I actually have one more question about **motorPowerSet**  parameter. For 4 motors I used different value to make the motor to start spin. M1 with value 19000 M2,M3,M4 with value 7000. Is this normal? Because I saw the post about the parameter around 3000 is enough to make the motor to spin. 
Is 4 motors should have same parameter value? When I use gamepad, the drone flipped, so I am wondering 4 motors require different parameter can be the potential reason that cause this issue. (I already check the propeller and motor spin direction).
